### PR TITLE
nspawn --private-users parsing, v2

### PR DESCRIPTION
**Pull request by keszybz**
Follow up for #4319.

### CODE REVIEW

File: src/nspawn/nspawn.c [hunk 1]
```diff
@@ -898,15 +898,21 @@ static int parse_argv(int argc, char *argv[]) {
...
 
-                case ARG_PRIVATE_USERS:
+                case ARG_PRIVATE_USERS: {
+                        int boolean = -1;
 
-                        r = optarg ? parse_boolean(optarg) : 1;
-                        if (r == 0) {
+                        if (!optarg)
+                                boolean = true;
+                        else if(strspn(optarg, DIGITS) != strlen(optarg))
```
Comment (poettering) [inline src/nspawn/nspawn.c]: There's in_charset() doing just this, but a tad bit more efficiently...

**Comment by poettering** [comment]
lgtm, though using in_charset() would be nicer

**Comment by keszybz** [comment]
Updated.

**Comment by poettering** [comment]
lgtm

**Pull request merged**